q2_amr/card/database.py

```diff
@@ -7,39 +7,53 @@
 import tempfile
 
 import requests
+from tqdm import tqdm
 
-from q2_amr.card.utils import run_command
+from q2_amr.card.utils import colorify, run_command
 from q2_amr.types._format import (
     CARDDatabaseDirectoryFormat,
     CARDKmerDatabaseDirectoryFormat,
 )
 
 
 def fetch_card_db() -> (CARDDatabaseDirectoryFormat, CARDKmerDatabaseDirectoryFormat):
-    # Fetch CARD and WildCARD data from CARD website
-    try:
-        response_card = requests.get(
-            "https://card.mcmaster.ca/latest/data", stream=True
-        )
-        response_wildcard = requests.get(
-            "https://card.mcmaster.ca/latest/variants", stream=True
-        )
-    except requests.ConnectionError as e:
-        raise requests.ConnectionError("Network connectivity problems.") from e
-
-    # Create temporary directory for WildCARD data
     with tempfile.TemporaryDirectory() as tmp_dir:
+        try:
+            card_tar_path = os.path.join(tmp_dir, "card_tar")
+            wildcard_tar_path = os.path.join(tmp_dir, "wildcard_tar")
+
+            # Download CARD and WildCARD tar database archives with progressbars
+            download_with_progressbar(
+                url="https://card.mcmaster.ca/latest/data",
+                progressbar_desc="Downloading CARD database",
+                tar_path=card_tar_path,
+            )
+
+            download_with_progressbar(
+                url="https://card.mcmaster.ca/latest/variants",
+                progressbar_desc="Downloading WildCARD database",
+                tar_path=wildcard_tar_path,
+            )
+
+        except requests.ConnectionError as e:
+            error_msg = "Unable to connect to the CARD server. Please try again later."
```

Comment:
No need for redundant variables - just use the msg directly when raising the error.